### PR TITLE
Search Context

### LINKED ISSUE #26: As a frequent user, I want to add links from feeds
so that I can save articles from a specific feed.

Part of #25

### LINKED ISSUE #109: As a user, I can search the web
so that I can find and add new topics without knowing specifics.

### Depends on:
- #26 adding links from feeds

### PR DESCRIPTION
**Pull request by qjqqyy**
This PR builds on top of #106 add-archives.
Part of #25.

### Summary of changes
* Add new context for viewing feed (resolves #26)
* Strip HTML tags from description field of the RSS feed when viewing.
* `bing` command searches bing for keywords (resolves #109)
* ~~Implement feed management (#69)~~ (#114)

### CODE REVIEW

**Comment by thomastanck** [comment]
> *    Add new context for viewing feed (resolves #26)
> *    Strip HTML tags from description field of the RSS feed when viewing.
> *    bing command searches bing for keywords (resolves #109)
> *    Implement feed management (#69) (WIP)

Is it possible to split this into 3-4 PRs?

**Comment by qjqqyy** [comment]
> Add new context for viewing feed (resolves #26)
> Strip HTML tags from description field of the RSS feed when viewing.
> bing command searches bing for keywords (resolves #109)
> Implement feed management (#69) (WIP)
> 
> 
> Is it possible to split this into 3-4 PRs?

1 and 3 are closely related, while 2 is a minor enhancement. Maybe I can extract the last one into a separate PR?

**Comment by thomastanck** [comment]
> 1 and 3 are closely related, while 2 is a minor enhancement. Maybe I can extract the last one into a separate PR?

Yep, sure! Then since 1-3 are done after extracting we can probably merge this in quickly.

File: src/main/java/seedu/address/commons/util/FeedUtil.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                new Address(DEFAULT_ADDRESS_TEXT),
+                Collections.emptySet()
+        );
+    }
+
+    /** Extracts a useful description from a SyndEntry. */
+    private static Description extractDescription(SyndEntry syndEntry, String feedUrl) {
+        String descriptionFromSyndEntry = syndEntry.getDescription().getValue();
+        String descriptionText = Jsoup.parseBodyFragment(descriptionFromSyndEntry).body().text()
+                .replace('\n', ' ').trim();
```
Comment (epicfailname) [inline src/main/java/seedu/address/commons/util/FeedUtil.java]: Do we have to consider replacing "\r" as well?

File: src/main/java/seedu/address/commons/util/FeedUtil.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                extractDescription(syndEntry, feedUrl),
+                new Link(syndEntry.getLink()),
+                new Address(DEFAULT_ADDRESS_TEXT),
+                Collections.emptySet()
+        );
+    }
+
+    /** Extracts a useful description from a SyndEntry. */
+    private static Description extractDescription(SyndEntry syndEntry, String feedUrl) {
+        String descriptionFromSyndEntry = syndEntry.getDescription().getValue();
```
Comment (epicfailname) [inline src/main/java/seedu/address/commons/util/FeedUtil.java]: From documentation of ```SyndEntry```, it says that ```getDescription()``` could return ```null``` if there is no description. Should we be more defensive here and ```assertNonNull()```?

**Review by epicfailname — DISMISSED** [review]

File: src/main/java/seedu/address/commons/util/FeedUtil.java (hunk 2 above)
Comment (qjqqyy) [reply src/main/java/seedu/address/commons/util/FeedUtil.java]: Great catch! The other `SyndEntry#getXXX` functions could also return `null` too (what a crappy library). Working on fixing it now!

Comment (qjqqyy) [reply src/main/java/seedu/address/commons/util/FeedUtil.java]: Should be guarded now! Thanks for the find!

File: src/main/java/seedu/address/commons/util/FeedUtil.java (hunk 1 above)
Comment (qjqqyy) [reply src/main/java/seedu/address/commons/util/FeedUtil.java]: Good consideration! I think that CRLF's should be taken care of at the program boundary but in any case this scratch snippet
```java
String desc = "<span>quick brown fox</span>\r\n<p>memes</p>";
Jsoup.parseBodyFragment(desc).body().text().;
```
behaves as expected which means Jsoup is probably also removing carriage returns. We should be safe here.

**Review by thomastanck — APPROVED** [review]
LGTM

**Pull request merged**